civl 1.7-3157 (new formula)

AlshammariA opened the pull request:
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

### Description

AlshammariA commented:
After I add things to the test part. I have a C sample example but I am not sure how can I add it. I feel comfortable with what I just add. I really need this software to publish soon that I might need your help.

UniqMartin commented:
> - [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?

You have ticked the checkbox, but it doesn't seem like you followed the steps outlined in there. I fixed the issue title now and will fix the commits when this gets merged, but wanted to point that out for future submissions you might want to make. Please also use the Git commit messages to describe the changes you've made. They are not the place to put comments that you should be posting on GitHub instead.

I guess you're simply not very experienced with Git and GitHub yet? Don't worry, we still appreciate your contribution, but you can make our life easier (and speed up acceptance of you submission) by trying to follow our guidelines. Thanks and I sincerely hope I haven't offended you.

UniqMartin commented on Formula/civl.rb:
```diff
@@ -0,0 +1,23 @@
+  end
+
+  test do
+    system "#{bin}/civl", "civl help", "--version", "civl gui"
```
What are you trying to do here? After installing `civl`, this is basically equivalent to entering the following on the command line (and I don't think this is what you want):

```
civl "civl help" --version "civl gui"
```

If I try that, I get:

```
CIVL v1.7 of 2016-03-31 -- http://vsl.cis.udel.edu/civl
Command line syntax error at [@0,0:3='civl',<14>,1:0]
Type "civl help" for command line syntax.
```

AlshammariA commented:
@UniqMartin Thank you for being patient with me during my submission. I have tried my best.

For what you get, does that mean that all the line has to be executed in one time. In fact, after we install civl, we type, "civl help" which gives us the options. This is used when we want to see if the civl is install or not.

AlshammariA commented:
I think I have to put 
system "#{bin}/civl", "help", "gui"
Instead of 
system "#{bin}/civl", "civl help", "--version", "civl gui"

Is that right?

UniqMartin commented:
> I have tried my best.

Thanks, that's appreciated!

> For what you get, does that mean that all the line has to be executed in one time. In fact, after we install civl, we type, "civl help" which gives us the options. This is used when we want to see if the civl is install or not.

Sorry, but I have a hard time understanding you. Can you try to rephrase what you are trying to say here? Also, if this is in response to my code comment, please add a comment there. This helps keep things organized and it makes it easier to understand what you are replying to.

UniqMartin commented:
> I think I have to put
> system "#{bin}/civl", "help", "gui"
> Instead of
> system "#{bin}/civl", "civl help", "--version", "civl gui"
> 
> Is that right?

Yes, if you use `system "#{bin}/civl", "help", "gui"` it will mean the same as typing `civl help gui` on the command line, so that's a command invocation that makes sense. Here's what I get with that:

```
CIVL v1.7 of 2016-03-31 -- http://vsl.cis.udel.edu/civl

Run the graphical interface of CIVL.

Usage: civl gui
```

This a very minimal test. We'd like to do better as we have discussed in #549.

DomT4 commented on Formula/civl.rb:
```diff
@@ -0,0 +1,23 @@
+  depends_on "cvc4"
+
+  def install
+    libexec.install "lib/civl-1.7_3157.jar"
```
To save some future work you might want to do something like:
```ruby
libexec.install "lib/civl-#{version}.jar"
bin.write_jar_script libexec/"civl-#{version}.jar", "civl"
share.install "doc", "examples", "licenses", "emacs"
```
Means won't have to be manually updated every time.

UniqMartin replied:
@DomT4 Note that this won't work as suggested because `version` is not equal to `1.7_3157`.

DomT4 replied:
Oh, arg, failed to spot that. That's a pain. Ignore me as desired then 😅.

AlshammariA replied:
@UniqMartin based on Civl documentation, I can run one of the example as a test. This is from the civl documentation:
_" To test your installation, copy the le examples/concurrency/locksBad.cvl to your work-
ing directory. Look at the program: it is a simple 2-process program with two shared variables
used as locks. The two processes try to obtain the locks in opposite order, which can lead
to a deadlock if both processes obtain their rst lock before either obtains the second. Type
\civl verify locksBad.cvl". You should see some output culminating in a message
The program MAY NOT be correct. See CIVLREP/locksBad_log.txt
"_

So do I need to write the command line since we already copy the files? Like this:

system "verify #{bin}/examples/concurrency/locksBad.cvl"

UniqMartin commented:
I merged your submission (with a minor correction) in a549b696bf0535e2b96d2b8cf3c97a317601315e. Because explaining what needs to be done would have been a bit complicated, I went ahead and added the test that is suggested in the Civl manual (and that you mentioned in another comment) in 8280d0f3a0b726d1a74c950f201be4d2af9e7895. I hope that's fine with you.

Have a look at this second commit to see what was necessary to make a good test. Feel free to ask if you have questions regarding the test code I added.

Thank you for this contribution to Homebrew, @AlshammariA! :tada:

AlshammariA commented:
I merged your submission (with a minor correction) in a549b69. Because explaining what needs to be done would have been a bit complicated, I went ahead and added the test that is suggested in the Civl manual (and that you mentioned in another comment) in 8280d0f. I hope that's fine with you.

Have a look at this second commit to see what was necessary to make a good test. Feel free to ask if you have questions regarding the test code I added.

Thank you for this contribution to Homebrew, @AlshammariA! :tada:

Thank you so much for being a great supporter. I did notice the changes which make meaningful for me. I will keep this in future updates of this formula. Thanks again.

AlshammariA commented:
> I merged your submission (with a minor correction) in a549b69. Because explaining what needs to be done would have been a bit complicated, I went ahead and added the test that is suggested in the Civl manual (and that you mentioned in another comment) in 8280d0f. I hope that's fine with you.
> 
> Have a look at this second commit to see what was necessary to make a good test. Feel free to ask if you have questions regarding the test code I added.
> 
> Thank you for this contribution to Homebrew, @AlshammariA!

Thank you so much for being a great supporter. I noticed the changes which are meaningful to me. Thanks again and wish the best for you.
